src/sentry/models/useravatar.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import absolute_import
+
+import uuid
+
+from PIL import Image
+
+from django.db import models
+
+from sentry.db.models import FlexibleForeignKey, Model
+from sentry.utils.cache import cache
+from sentry.utils.compat import StringIO
+
+
+class UserAvatar(Model):
+    """
+    A UserAvatar associates a User with their avatar photo File
+    and contains their preferences for avatar type.
+    """
+    __core__ = False
+
+    AVATAR_TYPES = (
+        (0, 'letter_avatar'),
+        (1, 'upload'),
+        (2, 'gravatar'),
+    )
+
+    ALLOWED_SIZES = (20, 48, 52, 64, 80, 96, 120)
+
+    user = FlexibleForeignKey('sentry.User', unique=True, related_name='avatar')
+    file = FlexibleForeignKey('sentry.File', unique=True, null=True, on_delete=models.SET_NULL)
+    ident = models.CharField(max_length=32, unique=True, db_index=True, null=True)
+    avatar_type = models.PositiveSmallIntegerField(default=0)
```

Comment:
if you make this have choices=AVATAR_TYPES you can (in get_avatar_type) do get_avatar_type_display, which is a magical Django thing